### PR TITLE
[REF] mass_mailing: clean JS views overrides

### PR DESCRIPTION
**Pull request by awa-odoo**
Task #1909424

The mass_mailing module used a lot of JS "include" to override
the behavior of the kanban view and the html_frame widget.

This commit cleans the overrides by using proper inheritance with "extend"
instead of "include".

It also slightly changes the current behavior of the mass_mailing kanban view
to disable the drag and drop feature only when the view is grouped by "state" or "email_from".
(Before this commit, the drag and drop feature was disabled for ALL fields).

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/kanban/kanban_renderer.js [hunk 1]
```diff
@@ -472,6 +473,21 @@ var KanbanRenderer = BasicRenderer.extend({
         });
         this.createColumnEnabled = this.groupedByM2O && this.columnOptions.group_creatable;
     },
+    /**
+     * Remove date/datetime magic grouping info to get proper field attrs/info from state
+     * ex: sent_date:month will become sent_date
+     *
+     * @private
+     * @param {string} groupByField
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/kanban/kanban_renderer.js]: missing
`@returns {string}`

**Comment by ged-odoo** [comment]
robodoo delegate=@awa-odoo

**Comment by awa-odoo** [comment]
robodoo r+

**Comment by awa-odoo** [comment]
robodoo rebase-ff

**Comment by awa-odoo** [comment]
@xmo-odoo Why can't I rebase-ff since I've been delegated the r+ rights on this PR? Seems weird.

@tde-banana-odoo Help plz thx bisoux.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
Je vais te péter les dents @robodoo

**Comment by d-fence** [comment]
Attention que parler à une machine c'est un peu de la régression.

**Comment by ged-odoo** [comment]
this one is interesting...

**Comment by xmo-odoo** [comment]
@awa-odoo it wasn't really clear what the needs would be, so I defaulted to doing the same as priorities (only intrinsic reviewers).

**Pull request closed without merging**